archive/imo/imo1969_q1.lean

```diff
@@ -0,0 +1,81 @@
+import tactic
+import tactic.basic
+import tactic.linarith
+import tactic.norm_cast
+import tactic.ring
+
+open int
+open nat
+
+/-
+The 1969 IMO, problem 1:
+Prove that there are infinitely many natural numbers $a$ with the following property:
+the number $z = n^4 + a$ is not prime for any natural number $n$.
+
+The key to the solution is that you can factor this into the product of two polynomials, if a = 4*m^4.
+-/
+
+lemma factorization (m n: ℤ): (n^2 + 2*m^2 - 2*n*m) * (n^2 + 2*m^2 + 2*n*m) = n^4 + 4*m^4
+:= by ring
+
+/-
+To show that the product is not prime, we need to show each of the factors is at least 2, which we can
+do with a sum-of-squares expression.
+-/
+lemma left_factor_large (m n: ℤ) (h: m > 1): (n^2 + 2*m^2 - 2*n*m) > 1 :=
+have h: (n^2 + 2*m^2 - 2*n*m) = (m-n)^2 + m^2, by ring,
+begin
+  rw h,
+  nlinarith,
+end
+
+lemma right_factor_large (m n: ℤ) (h: m > 1): (n^2 + 2*m^2 + 2*n*m) > 1 :=
```

Comment:
again these two >s should probably be <s (and several others below)

Reply:
ok